rfd/0215-bot-details-web.md

```diff
@@ -0,0 +1,218 @@
+
+![[0215-feature-active-instances.png]]
+
+#### Edit roles, traits and max session duration (`max_session_ttl`)
+Shows a dialog where the user can add and/or remove assigned roles, add and/or remove traits (well-known or custom), and edit the configured max session duration in the form `43200s`, `30m` or `3h`. Allow all changes to be made in a single atomic transaction.
+
+TODO: wireframe for edit form
+
+#### Delete bot
+Deletes the bot after confirmation. Shows a loading indicator during the call to the api. On success, navigates to the bots list (`/web/bots`). On error, shows a message within the confirmation dialog.
+
+![[0215-feature-delete-bot.png]]
+#### Lock bot
+Locks the bot after confirmation. Shows a loading indicator during the call to the api. On success, removes the dialog but remains on the bot detail page. On error, shows a message within the confirmation dialog.
+
+![[0215-feature-lock-bot.png]]
+### Implementation
+
+#### Data fetching and caching
+
+In order to keep the implementation modular, each logical section of the page fetches its own data, shows loading status and manages error states. This increases flexibility for future iterations while reducing the likelihood of introducing regression issues. Data caching behaviour can be tailored to each area of data independently.
+
+#### Web APIs
+
+| **New**                                         |                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
+| ----------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Fetch join tokens linked to a bot by name       | `GET /v1/webapi/sites/:site/machine-id/bot/:name/token`<br><br>**Logic**<br>Use `Service.ListResourcesWithFilter` to retrieve a paginated list of `token` resources which have a role of "Bot" and `bot_name` matching the bot being queried.<br><br>**Performance**<br>Pagination is not required, so pages will be retrieved one after the other until the end of the list. It is not anticipated that any bot will have more than 20 join methods. |
```

Comment:
Hmm, I was wondering if we ought to reuse the existing endpoint for listing Join Tokens, and just add a query parameter to filter down by associated Bot.

One challenge we may have here is I think the underlying auth server RPC for listing tokens requires an MFA tap, which can be pretty disruptive to UX. We'd either need to add a new RPC that doesn't require an MFA tap, and figure out what we'd need to redact to make that "Safe" or navigate the UX challenge here around MFA taps.

Reply:
That sounds pretty reasonable if we want to avoid introducing a new "redacted" ListTokens